src/components/ContentHighlights/data/hooks.js

```diff
@@ -143,3 +149,71 @@ export function useContentHighlightsContext() {
     setCatalogVisibilityAlert,
   };
 }
+
+
+export function useSetArchivedHighlightsCoursesCookies() {
+  const [highlightSet, setHighlightSet] = useState({});
+  const [isCookieLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const [cookieValue, setCookieValue] = useState({});
+
+  const { enterpriseCuration: { enterpriseCuration: { highlightSets } } } = useContext(EnterpriseAppContext);
+  const [isNewArchivedCourses, setIsNewArchivedCourses] = useState(false);
+
+  useEffect(() => {
+    const dismissedCookies = archivedHighlightsCoursesCookies.get(NEW_ARCHIVED_COURSE_ALERT_DISMISSED_COOKIE_NAME);
+
+    const isDismissedArchivedCoursesSet = dismissedCookies !== undefined;
+    const dismissedArchivedCourses = isDismissedArchivedCoursesSet ? dismissedCookies : {};
+
+    highlightSets.forEach(async(highlight) => {
+      try {
+        const { data } = await EnterpriseCatalogApiService.fetchHighlightSet(highlight.uuid);
+        const result = camelCaseObject(data);
+        setHighlightSet(result);
+      } catch (e) {
+        setError(e);
+        logError(e);
+      } finally {
+        setIsLoading(false);
+      }
+
+      const highlightedSetContent = highlightSet?.highlightedContent;
+      // checks that the highlight uuid isn't set
+      if (!dismissedArchivedCourses[highlight.uuid]) {
+        highlightedSetContent?.forEach(content => {
+          // if the length is greater than 1 for the course run status e.g. ["archived", "published"]
```

Comment:
I don't believe this is true, a course could have multiple archived course runs. The main thing is that for a course to be archived all of its course runs have to have that status as well.